packages/common/src/search/_internal/hubEventsHelpers/processFilters.ts

```diff
@@ -15,10 +15,24 @@ export function processFilters(filters: IFilter[]): Partial<GetEventsParams> {
   const processedFilters: Partial<GetEventsParams> = {};
   const access = getOptionalPredicateStringsByKey(filters, "access");
   if (access?.length) {
-    // TODO: remove ts-ignore once GetEventsParams supports filtering by access
-    // @ts-ignore
     processedFilters.access = access;
   }
+  const canEdit = getOptionalPredicateStringsByKey(filters, "canEdit");
+  if (canEdit?.length) {
+    processedFilters.canEdit = canEdit;
+  }
+  const entityIds = getOptionalPredicateStringsByKey(filters, "entityIds");
+  if (entityIds?.length) {
+    processedFilters.entityIds = entityIds;
+  }
+  const entityTypes = getOptionalPredicateStringsByKey(filters, "entityTypes");
+  if (entityTypes?.length) {
+    processedFilters.entityTypes = entityTypes;
+  }
+  const eventIds = getOptionalPredicateStringsByKey(filters, "eventIds");
+  if (eventIds?.length) {
+    processedFilters.eventIds = eventIds;
+  }
```

Comment:
While we're at it, I suspect we'll need the ability to filter for event results by specific creator ids at some point. I filed [10691](https://devtopia.esri.com/dc/hub/issues/10691) for the API to add a `createdById` filter parameter to the search events endpoint. Lets add support in preparation for that while we're in here. We'd likely want to re-use existing naming conventions from the front-end, i.e. `creator` while the actual API prop will probably more closely align with the `createdById` property defined on `IEvent`.

```ts
  const owner = getOptionalPredicateStringsByKey(
    filters,
    'owner',
  );
  if (owner?.length) {
    // TODO: remove below @ts-ignore once https://devtopia.esri.com/dc/hub/issues/10691 is completed
    // @ts-ignore
    processFilters.createdById = owner;
  }
```